Update bitcoin-cli to 0.21.1

johnongit opened the pull request:
Bitcoin-cli deployed in the docker image is a bit old (0.18.1).
We can update it to 0.21.1

vincenzopalazzo commented on Dockerfile:
```diff
@@ -17,7 +17,7 @@ RUN wget -qO /opt/tini "https://github.com/krallin/tini/releases/download/v0.18.
     && chmod +x /opt/tini
 
-ARG BITCOIN_VERSION=0.18.1
+ARG BITCOIN_VERSION=0.21.1
```
```suggestion
ARG BITCOIN_VERSION=0.22
```

Why not upgrade to the last version at this point?

johnongit replied:
Thx for your response.
I have not upgraded to 22.0 because the signature verification process is pretty different.
I can propose a new PR with the last version

vincenzopalazzo replied:
I don't see any reason to make the dev life more complex, I'm happy with v0.21.1 :) but also I'm happy to review your PR for 22.0 support :)

vincenzopalazzo reviewed (changes requested):
Hey thanks for your contribution  :)

Great work, I have only a small question that it is not a change request, but I would like to have also your opinion.

the only thing that I would like to ask is if you can squash the comment in the only one, if you are new with git [here](https://stackoverflow.com/questions/35703556/what-does-it-mean-to-squash-commits-in-git) there is a description

johnongit commented:
This my new PR. I have squashed my commits and updated bitcoin-cli to 22.0.
As you can see, SHA256SUMS.asc has been renamed to SHA256SUMS from https://bitcoincore.org/bin/bitcoin-core-22.0/.
Thanks for your review.

vincenzopalazzo reviewed (approved):
LGTM

utack https://github.com/ElementsProject/lightning/pull/4922/commits/4cf314981c24011ce0301758357867c903555043

cdecker commented:
ACK https://github.com/ElementsProject/lightning/pull/4922/commits/4cf314981c24011ce0301758357867c903555043
